crates/astria-conductor/src/executor/mod.rs

```diff
@@ -222,6 +229,29 @@ impl Executor {
             return Ok(None);
         }
 
+        match self
+            .executable_block_height
+            .cmp(&(block.header.height.value() as u32))
+        {
+            std::cmp::Ordering::Less => {
+                debug!(
+                    sequencer_block_height = block.header.height.value(),
+                    executable_block_height = self.executable_block_height,
+                    "skipping execution of future block"
+                );
+                return Ok(None);
+            }
+            std::cmp::Ordering::Greater => {
+                debug!(
+                    sequencer_block_height = block.header.height.value(),
+                    executable_block_height = self.executable_block_height,
+                    "skipping execution of stale block"
+                );
+                return Ok(None);
+            }
```

Comment:
either return an error, or update the function comment `/// if there are no relevant transactions in the SequencerBlock, it returns None.`

i think this should be an error, as `execute_block` shouldn't be called with a block that can't be executed